app/routines/update_user_contact_info.rb

```diff
@@ -123,6 +123,8 @@ def call
       if school.nil? && !sf_school.nil?
         users_without_cached_school += 1
         Sentry.capture_message("User #{user.id} has a school that is in SF but not cached yet #{sf_school.id}")
+      elsif sf_contact.school_id != user.school.salesforce_id
+        Sentry.capture_message("User #{user.id} has a mismatched school id (Salesforce: #{sf_contact.school_id}, Accounts: #{user.school.salesforce_id})")
```

Comment:
The previous behavior was `user.school = school` which is in the `else` below. We should probably still do this right? The other fields like `user.school_type` and `user.school_location` were already updated above.

Reply:
This is here to identify a problem that is happening when schools are merged in Salesforce.
I went through and fixed a bunch of them, and I'm not sure how frequent it happens (so logging for now).
The sync code [deletes the school properly](https://github.com/openstax/accounts/blob/main/app/routines/update_school_salesforce_info.rb#L29-L41) if it has no users, but if there are users and duplicate schools, it creates both - and the users are never moved from the merged school (hope that makes sense).

Reply:
What I'm saying is... because this overrides the stuff in the `else`, the user's school will not get updated here but some school-related fields were updated above so everything will be half-updated.

Reply:
Ah, gotcha. Thanks for pointing that out. 
See what you think of the change I made, I think it'll fix the broken records (and log for now, because I'm interested in how frequently this is happening)